Tests/js/views/services/assets/ez-roleserversideviewservice-tests.js

```diff
@@ -0,0 +1,609 @@
+            );
+        },
+    });
+
+    assignRoleTest = new Y.Test.Case({
+        name: "eZ Role Server Side View Service assign role test",
+
+        setUp: function () {
+            this.id = 'nyarlathotep';
+            this.contentId = '31';
+            this.contentName = 'The Crawling Chaos';
+            this.userId = 'azathoth';
+            this.userGroupId = 'yog-sothoth';
+            this.roleId = '42';
+            this.roleName = 'Shub-niggurath';
+            this.capi = new Mock();
+            this.userService = new Mock();
+            this.discoveryService = new Mock();
+            this.service = new Y.eZ.RoleServerSideViewService({
+                capi: this.capi
+            });
+            Mock.expect(this.capi, {
+                method: 'getUserService',
+                returns: this.userService,
+            });
+            Mock.expect(this.capi, {
+                method: 'getDiscoveryService',
+                returns: this.discoveryService,
+            });
+            Mock.expect(this.discoveryService, {
+                method: 'getInfoObject',
+                args: [Mock.Value.String, Mock.Value.Function],
+                run: function (infObjStr, cb) {
+                    if (infObjStr === 'users') {
+                        cb(false, { _href: '/api/ezp/v2/user/users' });
+                    } else if (infObjStr === 'rootUserGroup') {
+                        cb(false, { _href: '/api/ezp/v2/user/groups' });
+                    }
+                },
+            });
+        },
+
+        tearDown: function () {
+            delete this.service;
+            delete this.capi;
+            delete this.userService;
+        },
+
+        "Should assign the role to the discovered user": function () {
+            var that = this,
+                config = {},
+                content = new Mock(),
+                contentType = new Mock(),
+                id = this.id,
+                contentId = this.contentId,
+                contentName = this.contentName,
+                role = new Mock(),
+                roleId = this.roleId,
+                roleName = this.roleName,
+                roleAssignInputStruct = new Mock(),
+                universalDiscovery = new Mock(),
+                // TODO UDW should be limited to the Users subtree, and return User/UserGroup objects: EZP-24917
+                selection = [{content: content, contentType: contentType}],
+                callbackCalled = false,
+                loadRoleResponse = {
+                    document: {
+                        Role: role
+                    }
+                },
+                callback = function () {
+                    callbackCalled = true;
+                    Assert.areSame(
+                        that.service, this,
+                        "The callback should be executed in the service context"
+                    );
+                };
+
+            Mock.expect(content, {
```

Comment:
Yes, I know. Tried it, didn't work. I can make another attempt. (effing black magic)